database/sql/V033__new_table_pre_interactions.sql

```diff
@@ -0,0 +1,16 @@
+-- Creates new table for pre_interactions of orders
+-- Since one order can have many pre_interactions,
+-- the primary key is not the uid, but the whole 
+-- pre_interaction
+
+CREATE TABLE pre_interactions (
+    order_uid bytea NOT NULL,
+    target_to bytea NOT NULL,
+    value numeric(78,0) NOT NULL,
+    data bytea,
```

Comment:
Why?
I think some interactions don't need data. E.g. sending eth to a target.

I think we never check for callData.lenght> 0 here: https://github.com/cowprotocol/contracts/blob/main/src/contracts/libraries/GPv2Interaction.sol

Reply:
> I don't think so
agreed.
>that's why I think it should be not null.
I can't follow. We want to allow in theory that an empty byte array is sent over to the contract, right?

Reply:
Let's put it in Rust terms. We have a struct Interaction and it can contain either `data: Vec<u8>` or `data: Option<Vec<u8>>`. This is equivalent to `bytea NOT NULL` and `bytea` in Postgres. If we use `Option<Vec<u8>>` then `None` and `Some(vec![]))` have the same meaning. Either way we would create exactly the same transaction on chain. This is an indication that it is better to use the plain Vec rather than the Option. Because with the Option we have two ways to represent the same "no calldata" state. It is better to have only one way to represent the same state.